docs/getting-started/fundamentals/static-dynamic-rendering.md

```diff
@@ -0,0 +1,27 @@
+# Static or Dynamic rendering of a block
+
+
+
+A block can define dynamic rendering can be defined in two ways:
+    - The `register_block_type()` function includes a `render_callback` argument.
+    - A `render` property is added to `block.json`, whose value points to a separate PHP file (usually named `render.php`).
+
+Blocks can have static or dynamic rendering:
+- If no dynamic rendering has been been defined the rendering 
+
+## Blocks with Static Rendering 
+
+Blocks with static rendering **generate the HMTL for the frontend at "update-time" (when it's saved)**:
+- Blocks with static render use their markup representation in the DB to return the markup used in the frontend for the block
+- The `save()` function writes the block’s markup inside the block indicators (HTML comments). 
+- Only the markup inside the block indicators is returned as markup to be rendered for the block in the frontend.
+- For blocks with static render, the markup returned to the frontend is defined/set when the page is saved.
+
```

Comment:
Technically this is not entirely correct. Yes static blocks save their markup to the database and return said markup on the frontend. However both core and extenders can and do modify that markup before it gets rendered on the frontend via the `render_callback` or `render_block` hook. 

An example of this is the core cover block which is a static block that does some PHP magic if the "use featured image" setting to dynamically inject the featured image.

Same with the Image block. Yes it saves markup to the DB. But when the image is served through the media library the `sourceset` generation etc is done on the server. 

The List block is a static block that dynamically gets it's wrapping class name (`wp-block-list`) injected into the blocks markup on the server. 

--- 

My point here is not that this should all be documented here. Instead I worry that the current explanation makes one feel too "safe" that the markup generated in the `save` function in JS is output 100% as is without modification. Which is not true.

Reply:
> My point here is not that this should all be documented here. Instead I worry that the current explanation makes one feel too "safe" that the markup generated in the save function in JS is output 100% as is without modification. Which is not true.

Thanks for this explanation. Based on your explanations I agree the current text may be misleading.
I'll work on a different text proposal to take these cases into account.